account/src/middlewares/authStrategies.js

```diff
@@ -0,0 +1,61 @@
+import passport from 'passport';
+import jwt from 'jsonwebtoken';
+import { Strategy as LocalStrategy } from 'passport-local';
+import { Strategy as BearerStrategy } from 'passport-http-bearer';
+import Account from '../models/Account.js';
+import { decrypt } from '../utils/hash.js';
+import createCustomError from './customError.js';
+import { checkIfBlacklisted } from '../../redis/blacklistOperations.js';
+
+const verifyPassword = (password, hash) => {
+  if (!decrypt(password, hash)) {
+    return false;
+  }
+  return true;
+};
+
+const isTokenBlacklisted = async (token) => {
+  const result = await checkIfBlacklisted(token);
+  if (result) {
+    throw new jwt.JsonWebTokenError('Invalid token by logout');
+  }
+};
+
+passport.use(
+  new LocalStrategy({
+    usernameField: 'email',
+    passwordField: 'senha',
+    session: false,
+  }, (email, senha, done) => {
+    Account.findOne({ email }, (err, account) => {
+      if (err) {
+        return done(err);
+      }
+      if (account === null) {
+        return done(createCustomError(400, 'Invalid email or password'));
+      }
+      if (!verifyPassword(senha, account.senha)) {
+        return done(createCustomError(400, 'Invalid email or password'));
+      }
+      done(null, account);
+      return true;
+    });
+  }),
+);
+
+passport.use(
+  new BearerStrategy(
+    async (token, done) => {
+      try {
+        await isTokenBlacklisted(token);
```

Comment:
Em geral espera-se quem um método começando por "is" retorne um booleano, se quiser manter o lançamento da exception seria melhor um nome que não comece com is.